### PR TITLE
Generalizing safe workflows

### PR DESCRIPTION
**Pull request by Anabra**
Generalizing safe workflows
======================

Generalizing safe workflows with acyclic control-flows. Instead of having `XOR`, `Seq` and the special generalization of `XOR` called `GenXOR`, we introduce a new construct called _acyclic control-flow_. This is a construct which contains arrows and places. The transitions in the arrows can be any arbitrary safe workflow, just like in any other construct, but the arrows cannot loop nor go backwards.

This makes the model more expressive and also gets rid of the hidden loop inside `GenXOR`.

### CODE REVIEW

File: tests/Test/Workflow/SafeWorkflow/Examples.hs [hunk 1]
```diff
@@ -104,15 +99,23 @@ namedArbitraryNets = zipWith (\id net -> ("arbitrary-" <> show id, net)) [0..] a
...
+  , XOR (AND [Atom, Atom]) (XOR Atom Atom)
+  -- NOTE: cannot be expressed with GenACF
+  -- , Seq {seqLhs = GenLoop {gLoopIn = Nothing, gLoopExit = GenLoop {gLoopIn = Just Atom, gLoopExit = Atom, gLoopOut = Atom}, gLoopOut = GenXOR {gXorLhsIn = Atom, gXorLhsOut = Atom, gXorRhsIn = Atom, gXorRhsOut = Atom, gXorMToRhs = Just Atom, gXorMToLhs = Just Atom}}, seqRhs = XOR {xorLhs = GenLoop {gLoopIn = Just Atom, gLoopExit = Atom, gLoopOut = Atom}, xorRhs = GenLoop {gLoopIn = Just Atom, gLoopExit = Atom, gLoopOut = Atom}}}
+  -- TODO: reenable this
```
Comment (Acentelles) [inline tests/Test/Workflow/SafeWorkflow/Examples.hs]: Reenable this? :)

File: src/Data/List/List2.hs [hunk 2]
```diff
@@ -13,8 +14,10 @@ import GHC.Exts (IsList(..))
 
 import qualified Data.Foldable as F
 
+import Test.QuickCheck
+
 data List2 a = List2 a a [a]
```
Comment (Acentelles) [inline src/Data/List/List2.hs]: I like this. I think a short description clarifying that it's a list of at least 2 elements would help

File: tests/Test/Workflow/SafeWorkflow/Examples.hs (hunk 1 above)
Comment (Anabra) [reply tests/Test/Workflow/SafeWorkflow/Examples.hs]: Done.

File: src/Data/List/List2.hs (hunk 2 above)
Comment (Anabra) [reply src/Data/List/List2.hs]: Done.

**Pull request merged**